Boa.Constrictor/Screenplay/Caching/AnswerCache.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System.Collections.Generic;
+
+namespace Boa.Constrictor.Screenplay
+{
+    /// <summary>
+    /// Caches answers to Questions.
+    /// </summary>
+    public class AnswerCache
+    {
+        #region Instance Variables
+
+        /// <summary>
+        /// Lock object for safe multi-threading.
+        /// </summary>
+        protected readonly object Lock = new object();
+
+        /// <summary>
+        /// The answer cache.
+        /// Maps Question objects to answer objects.
+        /// </summary>
+        protected IDictionary<IInteraction, object> Cache { get; set; }
+
+        #endregion
+
+        #region Constructors
+
+        /// <summary>
+        /// Default constructor.
+        /// Initializes the cache to be empty.
+        /// </summary>
+        public AnswerCache()
+        {
+            Cache = new Dictionary<IInteraction, object>();
```

Comment:
Would it be better to remove the lock object and utilize the existing `ConcurrentDictionary`

Reply:
Unfortunately, the `public TAnswer Get<TAnswer>(IQuestion<TAnswer> question, IActor actor)` method below needs an explicit lock because it must be synchronized on the Question call, not only on the Dictionary access. At that point, using `ConcurrentDictionary` becomes unnecessary.